### PR TITLE
Drawer example for Dialog builder

### PR DESCRIPTION
**Pull request by Manuel-DaSilva**
Hello! In this PR I added a new example in the dialog builder page, it is a simple side drawer that acts as notification list. I think this is the first builder that has two examples, so I added a small explanation and separation between both of the examples. I am including the transition animation right in the component code, I believe this is good so the users can copy the full example and get the proper transition as seen in the docs.

I am attentive to how this can be improved and how should next examples should be built, already have a bottom-sheet example that would like to add too! Thanks

### CODE REVIEW

File: src/routes/docs/builders/dialog/drawer-code.ignore-svelte [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	<div use:portal>
+		{#if $open}
+			<div
+				{...$overlay}
+				class="fixed inset-0 z-20 bg-black/50"
+				transition:fade|local={{ duration: 150 }}
+			/>
+			<div
+				transition:slide|local={{ x: -350, duration: 200 }}
+				class="fixed left-0 top-0 z-[50] h-screen w-full max-w-[350px] bg-white p-[25px] shadow-lg
```
Comment (geerew) [inline src/routes/docs/builders/dialog/drawer-code.ignore-svelte:40]: `z-50` is built in so you don't need to use an arbitrary value

Comment (Manuel-DaSilva) [reply src/routes/docs/builders/dialog/drawer-code.ignore-svelte:40]: Thanks!

File: src/routes/docs/builders/dialog/drawer-code.ignore-svelte [hunk 2]
```diff
@@ -0,0 +1,80 @@
+<script lang="ts">
+	import { createDialog } from '@melt-ui/svelte';
+	import { fade } from 'svelte/transition';
+	import { expoOut } from 'svelte/easing';
+	import X from 'icons';
+
+	const { trigger, portal, overlay, content, title, description, close, open } = createDialog();
+
+	function slide(node: HTMLElement, options: { x: number; duration: number }) {
```
[the remark this answers is not in the document]
Comment (TGlide) [reply src/routes/docs/builders/dialog/drawer-code.ignore-svelte:9]: Doesn't slide offer up an option to remove opacity?

Also, I'm guilty of using a custom transition for dialog as well. Difference is, I didn't include the code for it anywhere, and exported it from helpers 😬 

Way I see it, there's a nice in-between. Use this custom fn in the preview, but use the normal slide in the copyable code..

[the remark this answers is not in the document]
Comment (geerew) [reply src/routes/docs/builders/dialog/drawer-code.ignore-svelte:9]: I like this approach from @TGlide. The example code should be clean and concise so as to make it a readable as possible

[the remark this answers is not in the document]
Comment (geerew) [reply src/routes/docs/builders/dialog/drawer-code.ignore-svelte:9]: And correct fly has an ops to y. Set it to 1 so it doesn’t fade ..

https://svelte.dev/docs#run-time-svelte-transition-fly

[the remark this answers is not in the document]
Comment (Manuel-DaSilva) [reply src/routes/docs/builders/dialog/drawer-code.ignore-svelte:9]: @geerew Yes you are right, maybe the transition improvement does not justify including the extra function. And @TGlide yes you are right! There is a opacity option, this makes my function useless and solves the issue directly :)

**Comment by TGlide** [comment]
While this is great, I'm waiting a bit before merging this as I'm working on refactoring the docs a bit!

**Comment by Manuel-DaSilva** [comment]
> While this is great, I'm waiting a bit before merging this as I'm working on refactoring the docs a bit!

Excellent! Will be attentive to integrate this example and many more :)

**Pull request merged**